Sources/Sentry/SentryUIViewControllerPerformanceTracker.m

```diff
@@ -320,6 +317,15 @@ - (void)viewControllerViewWillLayoutSubViews:(UIViewController *)controller
                 OBJC_ASSOCIATION_RETAIN_NONATOMIC);
         };
         [self.tracker activateSpan:spanId duringBlock:duringBlock];
+
+        // According to the Apple docs
+        // (https://developer.apple.com/documentation/uikit/uiviewcontroller/1621510-viewwillappear),
+        // viewWillAppear should be called for before the UIViewController is added to the view
+        // hierarchy. There are some edge cases, though, when this doesn't happen, and we saw
+        // customers' transactions also proofing this. Therefore, we must also report the initial
+        // display here, as the customers' transactions had spans for `viewWillLayoutSubviews`.
+
+        [self reportInitialDisplay:controller];
```

Comment:
m: What about `viewDidAppear` does it also have edge cases where it does not get called?
I would prefer it over `viewWillLayoutSubViews`

Reply:
The problem is that `viewDidAppear` could already be after a frame is rendered I guess.

Reply:
Actually many frames. It only trigger after the animation.